src/postproc/alerts.py

```diff
@@ -0,0 +1,227 @@
+# limitations under the License.
+
+import os
+import typing
+import numpy as np
+import glob
+import pathlib
+import pickle
+import gzip
+import itertools
+import multiprocessing
+
+import xarray as xr
+
+ALERTS_MINIMUM_DATA = 1 # nimimum data required to define alerts baseline
+
+def iterPickle(filename, compressed=True):
+    with gzip.open(filename) if compressed else open(filename, 'rb') as f:
+        while True:
+            try:
+                yield pickle.load(f)
+            except EOFError:
+                break
+
+def read_obs_file( path:  pathlib.Path,
+                   pop_keys: list =None,) -> list:
+    ''' read obs from file
+    remove keys specified by pop_keys if present.'''
+    result = [_ for _ in iterPickle( path)]
+    # throw away domain spec as first element
+    result.pop(0)
+    if pop_keys is not None:
+        for b in result:
+            for k in pop_keys:
+                b.pop(k)
+    return result
+
+def get_obs_sim(
+        dir: pathlib.Path | str,
+        obs_file_template: str,
+        sim_file_template: str,
+        ):
+    '''
+       reads obs and simulations from dir/obs_template_file and dir/sim_template_file,
+       checks for consistency of coordinates
+'''
+    obs_path = pathlib.Path.joinpath( pathlib.Path(dir), obs_file_template)
+    obs_list = read_obs_file( obs_path, pop_keys=['weight_grid'])
+    sim_path = pathlib.Path.joinpath( pathlib.Path(dir), sim_file_template)
+    sim_list = read_obs_file( sim_path, pop_keys=['weight_grid'])
+    if len(sim_list) != len( obs_list):
+        raise ValueError('inconsistent lenghts for obs and sim')
+    for obs, sim in zip( obs_list, sim_list):
+        if obs['lite_coord'] != sim['lite_coord']:
+            raise ValueError('inconsistent lite coord')
+    return obs_list, sim_list
+        
+def create_alerts_baseline(
+        domain_file: pathlib.Path,
+        dir_list: typing.Iterable,
+        obs_file_template: str = 'input/test_obs.pic.gz',
+        sim_file_template: str = 'simulobs.pic.gz',
+        output_file: str = 'alerts_baseline.nc',
+        ):
+    '''constructs a baseline for alerts.
+       the baseline consists of a mean and standard deviation for the differences between obs and simulation
+       at each point in the domain.
+       Output is stored as a netcdf file.
+       inputs:
+       domain_file: netcdf file describing the domain, will be used to template the output.
+       dir_list: list of directories containing obs and simulation outputs as ObservationData.
+       obs_file_template: string to be appended to each dir in dir_list to point to observations
+       sim_file_template: string to be appended to each dir in dir_list to point to simulations
+       output_file: name of output_file, will be overwritten if exists
+    '''
+    near_threshold = float( os.getenv('ALERTS_NEAR_THRESHOLD', '0.2'))
+    far_threshold = float( os.getenv('ALERTS_FAR_THRESHOLD', '1.0'))
```

Comment:
Reading these from the environment sort of breaks the "imperative shell / functional core" approach. Could they be function arguments with default values?

I also wonder whether it's important that these values are the same when creating alerts vs creating the baseline? Perhaps should they be added to the baseline output as attributes to make sure both equations use the same parameters?